patterns/nvidia-gpu-efa/generate-efa-nccl-test.sh

```diff
@@ -0,0 +1,141 @@
+            name: ${MPI_JOB_NAME}-launcher
+            imagePullPolicy: Always
+            env:
+             - name: LD_LIBRARY_PATH
+               value: "/opt/amazon/openmpi/lib:/opt/nccl/build/lib:/opt/amazon/efa/lib:/opt/aws-ofi-nccl/install/lib:/usr/local/nvidia/lib"
+             - name: PATH
+               value: "/opt/amazon/efa/bin:/usr/bin"
+             - name: XLA_FLAGS
+               value: "--xla_gpu_cuda_data_dir=/usr/local/cuda"
+             - name: TF_XLA_FLAGS
+               value: "--tf_xla_cpu_global_jit"
+             - name: NCCL_DEBUG
+               value: INFO
+            command:
+            - /opt/amazon/openmpi/bin/mpirun
+            - --allow-run-as-root
+            - --oversubscribe
+            - --tag-output
+            #- -N
+            #- "8"
+            - -np
+            - "${TOTAL_GPUS}"
+            - -bind-to
+            - none
+            - -map-by
+            - slot
+            - -x
+            - PATH
+            - -x
+            - LD_LIBRARY_PATH
+            - -x
+            - XLA_FLAGS
+            - -x
+            - TF_XLA_FLAGS
+            - -x
+            - NCCL_DEBUG=INFO
+            #- -x
+            #- NCCL_NVLS_ENABLE=${NCCL_NVLS_ENABLE}
+            #- -x
+            #- NCCL_PROTO=${NCCL_PROTO}
+            #- -x
+            #- NCCL_ALGO=${NCCL_ALGO}
+            #- -x
+            #- FI_PROVIDER=${FI_PROVIDER}
+            #- -x
+            #- FI_EFA_USE_DEVICE_RDMA=${FI_EFA_USE_DEVICE_RDMA}
+            #- -x
+            #- RDMAV_FORK_SAFE=${RDMAV_FORK_SAFE}
+            #- -x
+            #- NCCL_SHM_DISABLE=${NCCL_SHM_DISABLE}
+            - --mca
+            - pml
+            - ^cm
+            - --mca
+            - plm_rsh_agent
+            - ssh
+            - /opt/nccl-tests/build/all_reduce_perf
+            - -b
+            - "1"
+            - -e
+            - 2G
+            - -f
+            - "2"
+            - -t
+            - "1"
+            - -g
+            - "1"
+            - -c
+            - "1"
+            - -n
+            - "100"
+    Worker:
+      replicas: ${NUM_WORKERS}
+      template:
+        spec:
+          nodeSelector:
+            node.kubernetes.io/instance-type: "${INSTANCE_TYPE}"
+          containers:
+          - image: ${IMAGE_URI}
+            name: ${MPI_JOB_NAME}-worker
+            imagePullPolicy: Always
+            volumeMounts:
+            - name: shmem
+              mountPath: /dev/shm
+            resources:
+              limits:
+                nvidia.com/gpu: ${GPU_PER_WORKER}
+                hugepages-2Mi: ${HUGEPAGES_2MI}
+                vpc.amazonaws.com/efa: ${EFA_PER_WORKER}
+                memory: ${MEMORY}
+              requests:
+                nvidia.com/gpu: ${GPU_PER_WORKER}
+                hugepages-2Mi: ${HUGEPAGES_2MI}
+                vpc.amazonaws.com/efa: ${EFA_PER_WORKER}
+                memory: ${MEMORY}
+          volumes:
+          - name: shmem
```

Comment:
we are using shared memory or not? https://github.com/aws-ia/terraform-aws-eks-blueprints/pull/1966/files#diff-ad70f23fba6a87f2255bd64844bd763630ca4f67b559b37e09a65ed24b0415a9R17

Reply:
yes using shmem by default, removed setting to make it clear